File: Basics/While-Loops/Lab/graduation_pt2.py
student_name = input()
bad_appraisals = 0
appraisals_sum = 0
current_class_year = 1

while bad_appraisals < 2 and current_class_year <= 12:
    year_appraisal = float(input())

    if year_appraisal < 4:
        bad_appraisals += 1
        continue

    appraisals_sum += year_appraisal
    current_class_year += 1

if current_class_year <= 12:
    print(f"{student_name} has been excluded at {current_class_year} grade")
else:
    print(f"{student_name} graduated. Average grade: {appraisals_sum / 12:.2f}")

File: JS-Basics/0-5-While-Loops/Lab/graduation-pt-2.js
function graduation(input) {
    let i = 0; let grades = 0; let classN = 0; let excluded = 0;
    let name = input.shift(); input = input.map(n => Number(n));
    while (excluded < 2) {
        if (input[i] < 4) {excluded++; i++; continue;}
        classN++; grades += input[i];
        if (i == input.length - 1) {
            let average = grades / classN;
            return `${name} graduated. Average grade: ${average.toFixed(2)}`
        }
        i++;
    }
    classN++;
    return `${name} has been excluded at ${classN} grade`;  
}


console.log(graduation(["Gosho",
"5",
"5.5",
"6",
"5.43",
"5.5",
"6",
"5.55",
"5",
"6",
"6",
"5.43",
"5"]));
console.log(graduation(["Mimi",
"5",
"6",
"5",
"6",
"5",
"6",
"6",
"2",
"3"]));